docs/upgrade/xenial_upgrade_in_place.rst

```diff
@@ -0,0 +1,317 @@
+-----------------------------------------
+
+On your *Admin Workstation*, open a terminal by selecting
+**Applications > Favorites > Terminal**. Then connect to the
+*Application Server* with the command ``ssh app``.
+
+The operating system upgrade process should take a similar amount of time as
+the upgrade of the *Monitor Server*, and should not be interrupted once begun.
+
+To begin the process, run the command below and then supply the answers listed
+below to the prompts that follow.
+
+.. code:: sh
+
+  sudo do-release-upgrade
+
+.. note:: As with the *Monitor Server*, the exact prompts may vary based on your
+  hardware, and you should choose to preserve existing configuration files
+  whenever possible.
+
+- Run ``sudo do-release-upgrade`` to start the upgrade process
+- Press Enter after the "Some third party entries..." message
+- Type 'y' and press Enter to continue after the "Installing the upgrade..."
+  message
+- Type 'y' and press Enter to continue after the "Fetching and installing..."
+  message
+- Choose **OK** to close the Postfix information dialog
+- Choose **No Configuration** and **OK** on the "General type of mail
+  configuration" dialog
+- Choose **Yes** on the "Configuring libssl.0.0:amd64" dialog - when you do so,
+  the ``sshd`` and ``tor`` daemons will restart, breaking your connection to the
+  *Application Server*
+- Close the terminal window, open a new one, and reconnect with the command
+  ``ssh app``
+- Choose **No** to overriding local changes on the "PAM configuration" dialog
+- Choose the default action (keep local version) when prompted about
+  ``blacklist.conf`` changes
+- Choose **Keep the local version currently installed** on the
+  "Configuring grub-efi-amd64" dialog
+- Choose default (keep current version) when prompted about ``/etc/ssh/moduli``
+  changes
+- Choose default (keep current version) when prompted about
+  ``/etc/ssh/ssh_config`` changes
+- Choose default (keep current version) when prompted about ``/etc/pam.d/sshd``
+  changes
+- Choose **keep local version currently installed** and **OK** on the
+  "Configuring unattended-upgrades" dialog
+- Type 'y' and press Enter to remove obsolete packages when prompted
+- Type 'y' and press Enter to restart the system and complete the update
+
+The *Application Server* will now reboot - this may take several minutes. In
+order to reconnect via ``ssh app``, you must stop and restart the
+*Admin Workstation's* Internet connection,  using the upper-right-hand control
+in the Tails menu bar.
+
+To confirm that the upgrade succeeded, connect from a terminal using the command
+``ssh app`` and run the following command to display the installed OS version:
+
+.. code:: sh
+
+  sudo lsb_release -a
+
+The output should include the text "Ubuntu 16.04.5 LTS".
+
+Disconnect the SSH session to the Application Server. You are now ready to move
+on to the next step: reinstalling SecureDrop on the Xenial servers.
+
+Step 3: Reinstall SecureDrop 
```

Comment:
I propose changing the title to either re-provision or re-configure SecureDrop servers

Reply:
done!